contracts/SignHash.sol

```diff
@@ -26,9 +28,22 @@ contract SignHash {
         HashSigned(hash, msg.sender);
     }
 
+    function prove(string method, string value) {
+        require(bytes(method).length > 0);
+        require(bytes(value).length > 0);
+
+        signerProofs[msg.sender][method] = value;
+
+        SignerProved(msg.sender, method, value);
+    }
+
     //--- Public constant functions
 
     function getSigners(bytes32 hash) public constant returns (address[]) {
         return hashSigners[hash];
     }
+
+    function getProof(address signer, string method) public constant returns (string) {
+        return signerProofs[signer][method];
+    }
```

Comment:
AFAIR making the `signerProofs` public should be just enough and generate a `signerProof(signer, method)` getter itself.

Reply:
I'd rather not use autogenerated getters here. Solidity cannot generate getter for `hashSigners` field so I prefer to make both fields private and provide custom getters. I am also not sure if I like `signerProofs(signer, method)` signature, `getProof` sounds better to me.

Reply:
👍